src/services/impl/LeaderboardManager.cpp

```diff
@@ -1,14 +1,29 @@
-#include "RA_LeaderboardManager.h"
+#include "LeaderboardManager.hh"
+
 #include "RA_MemManager.h"
 #include "RA_PopupWindows.h"
 #include "RA_md5factory.h"
 #include "RA_httpthread.h"
 
+#include "services\ServiceLocator.hh"
+
 #include <ctime>
 
-RA_LeaderboardManager g_LeaderboardManager;
+namespace ra {
+namespace services {
+namespace impl {
+
+LeaderboardManager::LeaderboardManager()
+    : LeaderboardManager(ra::services::ServiceLocator::Get<ra::services::IConfiguration>())
+{
+}
```

Comment:
Overloaded constructor allows for lazy resolution of dependency. For illustrative purposes only. Will likely remove before merging.